components/datadog/agent/docker.go

```diff
@@ -98,12 +98,19 @@ func dockerAgentComposeContent(agentImagePath string, apiKey pulumi.StringInput,
 						"/var/run/docker.sock:/var/run/docker.sock",
 						"/proc/:/host/proc",
 						"/sys/fs/cgroup/:/host/sys/fs/cgroup",
+						"/var/run/datadog:/var/run/datadog",
 					},
 					Environment: map[string]any{
-						"DD_API_KEY":                     apiKeyResolved,
-						"DD_PROCESS_AGENT_ENABLED":       true,
-						"DD_DOGSTATSD_NON_LOCAL_TRAFFIC": true,
+						"DD_API_KEY":                           apiKeyResolved,
+						"DD_PROCESS_AGENT_ENABLED":             true,
+						"DD_DOGSTATSD_NON_LOCAL_TRAFFIC":       true,
+						"DD_DOGSTATSD_ORIGIN_DETECTION":        true,
+						"DD_DOGSTATSD_TAG_CARDINALITY":         "high",
+						"DD_DOGSTATSD_SOCKET":                  "/var/run/datadog/dsd.socket",
+						"DD_DOGSTATSD_ORIGIN_DETECTION_CLIENT": true,
 					},
+					Pid:   "host",
+					Ports: []string{"8125:8125/udp"},
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wonder if this should be an option, and if we should change this function's signature to a `DockerComposeManifestService` to merge the default one with. The env variables you add should be passed through `envVars`, using `WithAgentServiceEnvVars`, rather than having them as default

Reply:
I moved the env vars related to dsd into the docker-compose of the dsd apps